.github/workflows/tests.yml

```diff
@@ -16,7 +16,8 @@ jobs:
     - name: Install dependencies
       run: |
         curl -sSL https://install.python-poetry.org | python3
-        poetry install --all-extras
+        poetry lock
```

Comment:
I know poetry lock itself takes a relatively long time. Should we add this? My understanding is that poetry lock is for refreshing the profile and should be only done at each time we change the pyproject.toml

Reply:
Leaving it out creates problems in supporting multiple python versions. Let me know if you find solutions to this.

Reply:
ok, thansk for telling this.